lib/devise/strategies/multi_factor_authenticatable.rb

```diff
@@ -13,7 +13,7 @@ def authenticate!
         end
 
         if validate(resource) { hashed = true; resource.valid_password?(password) } # rubocop:disable Style/Semicolon
-          if resource.needs_mfa?
+          if resource.needs_mfa? || authentication_level_does_not_need_mfa?
             env["warden.mfa.bypass"] = MultiFactorAuth.verify_bypass_token(resource, env["warden.mfa.bypass_token"])
             env["warden.mfa.required"] = !env["warden.mfa.bypass"]
```

Comment:
I think rather than having this negative conditional it could be clearer to be more verbose, something like:

```ruby
if resource.needs_mfa?
  env["warden.mfa.bypass"] = MultiFactorAuth.verify_bypass_token(resource, env["warden.mfa.bypass_token"])
  env["warden.mfa.required"] =
    if env["warden.mfa.authenticate_to_level"] == "level0"
      false
    else
      !env["warden.mfa.bypass"]
    end
```

Reply:
sure I'm up for that!

Reply:
This does change the logic from an `or` to an `and`... with `needs_mfa?` but i think that's OK as in every other secenario the phone number's existance and these levels should be in lock-step, right?

I'll leave it in it's own method though because I still think it's worth having the feature flag?
Happy to hear push back if you disagree?

Reply:
Since we check the feature flag in the sessions controller, I don't think we also need to do it here.

Reply:
ok cool will fixup